tools/hack/create-cluster.sh

```diff
@@ -102,3 +138,29 @@ spec:
   ipAddressPools:
   - kube-services
 EOF
+}
+
+RETRY_INTERVAL=5  # seconds
+TIMEOUT=120        # seconds
+ELAPSED_TIME=0
+
+if [ "$CUSTOM_CNI" = "true" ]; then
```

Comment:
can you share more details about why we need cilium?

Reply:
The default network plugin in Kubernetes-in-Docker (Kind) does not support enforcing network policies that restrict connectivity to local services, such as the Kubernetes API server. However, by disabling the default CNI and deploying Cilium as the CNI plugin, we can implement network policies that effectively control access to the API server. This approach allows for precise management of network traffic within the cluster, including the ability to block connectivity to the Kubernetes API server as needed.

Reply:
I recall kind support network policy as https://github.com/kubernetes-sigs/kind/issues/842 closed.

Reply:
will https://github.com/kubernetes-sigs/kube-network-policies help?

Reply:
I will give it a try, maybe it can do the work :)

Reply:
@zirain I tested with kube-network-policies, but it was unable to block connectivity to the API server. In fact, I tried all the other available options as well. Currently, it seems this functionality is only achievable with the Cilium CNI.